### PR TITLE
Add Django 4.1 constraints violation_error_message

### LINKED ISSUE #1262: Django 4.1 constraints violation_error_message
Hi there,

I dont think there is an implementation of the new https://docs.djangoproject.com/fr/4.1/ref/models/constraints/#violation-error-message on constraints types.

Is there another issue tracking this that I'm missing? Or is that okay if I work on this?

### PR DESCRIPTION
**Pull request by martinlehoux**
# I have made things!

- https://docs.djangoproject.com/fr/4.1/ref/models/constraints/#violation-error-message

## Related issues

- Closes https://github.com/typeddjango/django-stubs/issues/1262

### CODE REVIEW

File: django-stubs/db/models/constraints.pyi [hunk 1]
```diff
@@ -14,7 +14,8 @@ class Deferrable(Enum):
 
 class BaseConstraint:
     name: str
-    def __init__(self, name: str) -> None: ...
+    violation_error_message: str | None = ...
```
Comment (sobolevn) [inline django-stubs/db/models/constraints.pyi]:
```suggestion
    violation_error_message: str | None
```

File: tests/typecheck/db/models/test_constraints.yml [hunk 2]
```diff
@@ -35,3 +35,13 @@
         main:4: note: Possible overload variants:
         main:4: note:     .*
         main:4: note:     .*
+
+-   case: constraint_violation_error_message
```
Comment (sobolevn) [inline tests/typecheck/db/models/test_constraints.yml]: We don't need tests for simple things. We only test plugins / complex overloads / etc

**Review by sobolevn — COMMENTED** [review]
Thanks!

File: tests/typecheck/db/models/test_constraints.yml [hunk 3]
```diff
@@ -35,3 +35,13 @@
...
+    main: |
+        from django.db.models import CheckConstraint, Q
+
+        CheckConstraint(
+            check=Q(name__startswith='a'),
+            name='name_starts_with_a',
+            violation_error_message='Name must start with a',
+        )
```
Comment (adamchainz) [inline tests/typecheck/db/models/test_constraints.yml]:
```suggestion
```

File: tests/typecheck/db/models/test_constraints.yml (hunk 2 above)
Comment (adamchainz) [reply tests/typecheck/db/models/test_constraints.yml]: Agree

**Review by adamchainz — APPROVED** [review]
LGTM after fixes

**Comment by martinlehoux** [comment]
Thanks for pushing this haha

**Pull request merged**